web-marketplace/src/pages/BuyCredits/BuyCredits.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMemo, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useLingui } from '@lingui/react';
+
+import { useFetchSellOrders } from 'features/marketplace/BuySellOrderFlow/hooks/useFetchSellOrders';
+import WithLoader from 'components/atoms/WithLoader';
+import { CardSellOrder } from 'components/organisms/ChooseCreditsForm/ChooseCreditsForm.types';
+import { MultiStepTemplate } from 'components/templates/MultiStepTemplate';
+import { useGetProject } from 'components/templates/ProjectDetails/hooks/useGetProject';
+
+import { PAYMENT_OPTIONS } from './BuyCredits.constants';
+import { BuyCreditsForm } from './BuyCredits.Form';
+import { PaymentOptionsType } from './BuyCredits.types';
+import { getFormModel } from './BuyCredits.utils';
+
+export const BuyCredits = () => {
+  const { _ } = useLingui();
+  const {
+    sanityProject,
+    projectBySlug,
+    loadingProjectBySlug,
+    projectByOnChainId,
+    loadingProjectByOnChainId,
+    offchainProjectByIdData,
+    loadingOffchainProjectById,
+    isBuyFlowDisabled,
+    projectsWithOrderData,
+    onChainProjectId,
+    offChainProject,
+    creditClassOnChain,
+    loadingBuySellOrders,
+  } = useGetProject();
+
+  const [paymentOption, setPaymentOption] = useState<PaymentOptionsType>(
+    sanityProject?.fiatSellOrders && sanityProject?.fiatSellOrders.length > 0
+      ? PAYMENT_OPTIONS.CARD
+      : PAYMENT_OPTIONS.CRYPTO,
+  );
+  const [retiring, setRetiring] = useState<boolean>(true);
+
+  const formModel = getFormModel({ _, paymentOption, retiring });
+  const sellOrders = useMemo(
+    () => projectsWithOrderData?.[0]?.sellOrders || [],
+    [projectsWithOrderData],
+  );
+
+  const cardSellOrders = useMemo(
+    () =>
+      sanityProject?.fiatSellOrders?.map(fiatOrder => ({
+        ...fiatOrder,
+        ...sellOrders.filter(
+          cryptoOrder => cryptoOrder.id.toString() === fiatOrder?.sellOrderId,
+        )?.[0],
+      })) || [],
+    [sanityProject?.fiatSellOrders, sellOrders],
+  );
```

Comment:
This could be refactored to improve readability by extracting it into its own function:

```
const getCardSellOrders = (fiatSellOrders, sellOrders) => {
  return fiatSellOrders?.map(fiatOrder => ({
    ...fiatOrder,
    ...sellOrders.find(
         cryptoOrder => cryptoOrder.id.toString() === fiatOrder?.sellOrderId,
    ),
  })) || [];
};

const cardSellOrders = useMemo(
  () => getCardSellOrders(sanityProject?.fiatSellOrders, sellOrders),
  [sanityProject?.fiatSellOrders, sellOrders]
);
```